### PR TITLE
fix: #1206 #1207 clean up and add new spar roles

### PR DESCRIPTION
**Pull request by MCatherine1994**
refs: #1206

The role description got from Sabina is as below, but some of them is longer than 200 chars, so I made a little bit change: 

**TSC_ADMIN** user role allows a ministry SPAR user to access SPAR administrative screens and perform administrative functions such as seedlot registration approval, or seedling request approval

**SPAR_SPR_MINISTRY_ORCHARD** user role allows ministry SPAR users to access the A class registration screens and to create, review, update and submit A class registration forms.
Ministry users may enter and update lots within their own Organizational Unit. 

**SPAR_SPR_NONMINISTRY_ORCHARD** user role allows non ministry SPAR users to access the A class registration screens and to create, review, update and submit A class registration forms.
Non-Ministry Users may enter and update lots where their sign in client profile matches the Applicant Client.

SPAR team is okay to use the role name `SPAR_TSC_ADMIN`, `SPAR_MINISTRY_ORCHARD` and `SPAR_NONMINISTRY_ORCHARD` for consistency

### CODE REVIEW

File: server/flyway/sql/V46__add_spar_roles.sql [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+-- Add roles for SPAR_DEV
+INSERT INTO app_fam.fam_role (
+    role_name,
+    role_purpose,
+    application_id,
+    role_type_code,
+    create_user,
+    create_date
+)
+VALUES ('SPAR_TSC_ADMIN', 'Allow ministry users to access SPAR administrative screens and perform administrative functions such as seedlot registration approval, or seedling request approval', (select application_id from app_fam.fam_application where application_name = 'SPAR_DEV'), 'C', CURRENT_USER, CURRENT_DATE),
```
Comment (ianliuwk1019) [inline server/flyway/sql/V46__add_spar_roles.sql:22]: Maybe we can utilize the `with` subquery for this application_id: 
```
    WITH spar_dev_application_id AS (select application_id from app_fam.fam_application where application_name = 'SPAR_DEV')
```
to reduce the inline query for clarity.
V38_ flyway has example.

File: server/flyway/sql/V46__add_spar_roles.sql [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+-- Add roles for SPAR_TEST
+INSERT INTO app_fam.fam_role (
+    role_name,
+    role_purpose,
+    application_id,
+    role_type_code,
+    create_user,
+    create_date
+)
+VALUES ('SPAR_TSC_ADMIN', 'Allow ministry users to access SPAR administrative screens and perform administrative functions such as seedlot registration approval, or seedling request approval', (select application_id from app_fam.fam_application where application_name = 'SPAR_TEST'), 'C', CURRENT_USER, CURRENT_DATE),
```
Comment (ianliuwk1019) [inline server/flyway/sql/V46__add_spar_roles.sql:36]: Hi Catherine, just want to confirm, should this be "**TSC_ADMIN**" or "**SPAR_**TSC_ADMIN"?
The description seems to only mentioned "**TSC_ADMIN**" is the one they need without **"SPAR_"**.

**Comment by ianliuwk1019** [comment]
Hi @MCatherine1994 ,
So this will delete "ALL" spar roles current in database.
```
DELETE FROM app_fam.fam_role WHERE application_id IN (
    SELECT application_id FROM app_fam.fam_application WHERE application_name IN ('SPAR_DEV', 'SPAR_TEST', 'SPAR_PROD')
);
```
Just want to confirm, that's what all they don't need?

File: server/flyway/sql/V46__add_spar_roles.sql (hunk 2 above)
Comment (MCatherine1994) [reply server/flyway/sql/V46__add_spar_roles.sql:36]: yeah we had a meeting with them this morning, and they are fine with our naming.  SPAR_TSC_ADMIN, SPAR_MINISTRY_ORCHARD and SPAR_NONMINISTRY_ORCHARD, I put in the description as well

**Comment by MCatherine1994** [comment]
> Hi @MCatherine1994 , So this will delete "ALL" spar roles current in database.
> 
> ```
> DELETE FROM app_fam.fam_role WHERE application_id IN (
>     SELECT application_id FROM app_fam.fam_application WHERE application_name IN ('SPAR_DEV', 'SPAR_TEST', 'SPAR_PROD')
> );
> ```
> 
> Just want to confirm, that's what all they don't need?

yeah we mentioned this to them this morning, they only need 3 roles for next release. And for the rest of roles, their name could change in the future as well. We'll do a one time cleanup, and they will use the 3 new roles to grant access as they need.

File: server/flyway/sql/V46__add_spar_roles.sql (hunk 1 above)
Comment (MCatherine1994) [reply server/flyway/sql/V46__add_spar_roles.sql:22]: yeah... I actually followed the v24 where last time we updated the spar roles, also like v39. I'm also fine to update it, both work for me.

File: server/flyway/sql/V46__add_spar_roles.sql (hunk 2 above)
Comment (ianliuwk1019) [reply server/flyway/sql/V46__add_spar_roles.sql:36]: Got it.

File: server/flyway/sql/V46__add_spar_roles.sql (hunk 1 above)
Comment (ianliuwk1019) [reply server/flyway/sql/V46__add_spar_roles.sql:22]: Sure, can leave it.

**Review by ianliuwk1019 — APPROVED** [review]

**Pull request merged**